uniprot-search/src/main/java/org/uniprot/store/search/field/validator/FieldRegexConstants.java

```diff
@@ -13,6 +13,9 @@
 public class FieldRegexConstants {
     public static final String UNIPROTKB_ACCESSION_REGEX =
             "([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?";
+
+    public static final String UNIPROTKB_ACCESSION_OR_ID = "([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])((-[0-9]{1,3})|(\\.[0-9]{1,3})|(_[A-Z0-9]{2,5}))?|[A-Z0-9]{2,5}_[A-Z0-9]{2,5}";
+
     public static final String PROTEOME_ID_REGEX = "UP[0-9]{9}";
```

Comment:
First line of the review is so complicated :-D 
So the version can only have value from 0 to 999? I am ignoring the case where this regex can pass version 0

Reply:
Yes, currently max version are less than 300, so it will last at least 700 months